### PR TITLE
Prepare Flet v0.24.0

### PR DESCRIPTION
**Pull request by ndonkoHenri**
## Summary by Sourcery

Prepare for Flet v0.24.0 by introducing new features such as enhanced audio state handling, new theme customization options, and improved event handling across controls. Refactor theme classes for better code clarity and update various controls with new properties for enhanced customization. Fix enum naming issues and ensure proper handling of blend_mode in relevant controls. Update the Flet CLI environment variable for disabling rich output and enhance test coverage for new features.

New Features:
- Introduce new classes for handling audio state changes, including AudioState, AudioStateChangeEvent, AudioPositionChangeEvent, and AudioDurationChangeEvent.
- Add new DataTableTheme and ButtonTheme classes to enhance theme customization.
- Implement new event handling capabilities for various controls, allowing for more granular control over events like on_click, on_change, and on_animation_end.
- Add support for new properties in CupertinoTextField, such as image, enable_scribble, padding, scroll_padding, and obscuring_character.
- Introduce new properties for Tooltip, including blend_mode, shadow, and image, to enhance customization.

Bug Fixes:
- Fix incorrect enum naming for MarkdownSelectionChangeCause and BorderSideStrokeAlign.
- Correct the handling of blend_mode in Container and Tooltip to ensure it only applies when a background color or gradient is provided.

Enhancements:
- Refactor theme-related classes to remove the use of field(default=None) in favor of direct assignment to None.
- Improve the CupertinoTextField by adding padding and scroll padding parsing.
- Enhance the Tooltip control with a new boxDecorationFromDetails function for better decoration handling.
- Refactor event handling in various controls to use OptionalControlEventCallable for better type safety and consistency.
- Update the CupertinoTextField to include a reveal password icon with padding for better UI consistency.

Build:
- Update the Flet CLI to use the environment variable FLET_CLI_NO_RICH_OUTPUT instead of FLET_NO_RICH_OUTPUT for disabling rich output.

Tests:
- Add tests for blend_mode in Container to ensure proper functionality when a background color is provided.
- Update Tooltip tests to include shadow attribute verification.

Chores:
- Deprecate the use of field(default=None) in dataclass fields across multiple theme-related classes.

### CODE REVIEW

**Comment by Muddassirofficial** [comment]
Where is ads supports?

**Comment by ndonkoHenri** [comment]
Here: https://github.com/flet-dev/flet/pull/3288 :)

File: sdk/python/packages/flet-core/src/flet_core/expansion_tile.py [hunk 1]
```diff
@@ -63,8 +62,10 @@ def __init__(
         collapsed_shape: Optional[OutlinedBorder] = None,
         dense: Optional[bool] = None,
         enable_feedback: Optional[bool] = None,
+            show_trailing_icon: Optional[bool] = None,
+            min_tile_height: OptionalNumber = None,
         visual_density: Union[None, ThemeVisualDensity, VisualDensity] = None,
-        on_change: OptionalEventCallable = None,
+            on_change: OptionalControlEventCallable = None,
         #
```
Comment (FeodorFitsner) [inline sdk/python/packages/flet-core/src/flet_core/expansion_tile.py:69]: Indentation slipped.

File: sdk/python/packages/flet-core/src/flet_core/menu_item_button.py [hunk 2]
```diff
@@ -36,10 +36,13 @@ def __init__(
...
-        on_focus: OptionalEventCallable = None,
-        on_blur: OptionalEventCallable = None,
+            semantic_label: Optional[str] = None,
+            autofocus: Optional[bool] = None,
+            overflow_axis: Optional[Axis] = None,
+            on_click: OptionalControlEventCallable = None,
+            on_hover: OptionalControlEventCallable = None,
+            on_focus: OptionalControlEventCallable = None,
+            on_blur: OptionalControlEventCallable = None,
         #
```
Comment (FeodorFitsner) [inline sdk/python/packages/flet-core/src/flet_core/menu_item_button.py:46]: Indentation.

File: sdk/python/packages/flet-core/src/flet_core/page.py [hunk 3]
```diff
@@ -2770,7 +2771,7 @@ def on_resized(self):
         return self.__on_resized
 
     @on_resized.setter
-    def on_resized(self, handler: "Optional[Callable[[WindowResizeEvent], None]]"):
+    def on_resized(self, handler: OptionalEventCallable["WindowResizeEvent"]):
         self.__on_resized.subscribe(handler)
```
Comment (FeodorFitsner) [inline sdk/python/packages/flet-core/src/flet_core/page.py:2775]: Event should be updated to a new implementation.

File: sdk/python/packages/flet-core/src/flet_core/tabs.py [hunk 4]
```diff
@@ -157,14 +157,20 @@ def __init__(
...
         indicator_thickness: OptionalNumber = None,
         enable_feedback: Optional[str] = None,
         mouse_cursor: Optional[MouseCursor] = None,
+            padding: PaddingValue = None,
+            splash_border_radius: BorderRadiusValue = None,
         clip_behavior: Optional[ClipBehavior] = None,
-        on_change: OptionalEventCallable = None,
+            on_click: OptionalControlEventCallable = None,
+            on_change: OptionalControlEventCallable = None,
         #
```
Comment (FeodorFitsner) [inline sdk/python/packages/flet-core/src/flet_core/tabs.py:174]: Indentation.

**Review by Muddassirofficial — APPROVED** [review]

**Review by FeodorFitsner — APPROVED** [review]

**Pull request merged**